src/PrestaShopBundle/ApiPlatform/Provider/QueryListProvider.php

```diff
@@ -88,24 +88,25 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
             );
         }
 
-        $queryBuilderDefinition = $operation->getExtraProperties()['queryBuilder'] ?? null;
+        $gridDataFactoryDefinition = $operation->getExtraProperties()['gridDataFactory'] ?? null;
 
-        if (null === $queryBuilderDefinition) {
-            throw new QueryBuilderNotFoundException(sprintf('Resource %s has no Query builder defined.', $operation->getClass()));
+        if (null === $gridDataFactoryDefinition) {
+            throw new GridDataFactoryNotFoundException(sprintf('Resource %s has no Grid data factory defined.', $operation->getClass()));
         }
 
-        /** @var DoctrineQueryBuilderInterface $queryBuilder * */
-        $queryBuilder = $this->container->get($queryBuilderDefinition);
+        /** @var GridDataFactoryInterface $gridDataFactory */
+        $gridDataFactory = $this->container->get($gridDataFactoryDefinition);
```

Comment:
I forgot about this one, but in a later PR we could refacto and improve this a bit, we are relying on the global container meaning all grid data factory services need to be public to be used here We have the same issue in controllers of course but that's something that could be avoided by relying on Symfony DI tools like service containers, tagged iterators and so on... But not relevant for this PR anyway